### ISSUE TITLE
Mouse moves twitchy with pressed button (while selecting text, for example)

### ISSUE THREAD

**Issue by roussak** [FreeRDP/FreeRDP#6919]
I used the freerdp v.2.2.0 package in the Arch Linux. I use it to connect to my work computer under the Windows 10. With the 2.2.0 it worked OK. Then I upgraded all software on my laptop. The freerdp package was upgraded to the 2.3.2. With the new version, the MS Word (for example) using became a bit tricky.
It is manifested in the following. Let's say, there is text in the MS Word. This editor has a feature: selecting text by words. But this feature works a bit complex. Let's say, you move the mouse pointer to position that corresponds to the middle of some word in text, then you press the mouse button and move it again. I.e., you perform text selection. At the very beginning stage of this process, MS Word selects part of the word starting from the position where you was clicked. Then, if you continue movement and mouse pointer goes out of the word, MS Word selects this word entirely (and continue a text selection if you continue to move pointer). And here is the thing: if you move the pointer back to the word you started with, then MS Word cancels the word's selection as if this feature is switched off.
Now, about  the freerdp... )) It behaves as if I moved mouse pointer somewhere back during selection (i.e., while button is pressed). It is not appear always. It is not necessary that it move the pointer to the position somewhere near of the starting one. But, because the word selection feature works as I described here, its using became tricky. Because, let's say, you started a text selection, MS Word selected some word and... freerdp twitched the pointer back and you didn't get exactly what you expected -- text now is selected from the middle of the word.
When I selected cells in a table, the freerdp sometimes behaves in a similar way: cells selection flickers, its boundaries are changed in unexpected manner.
I noticed, that this issue is reproduced more "willingly" when mouse movement is faster during selection.
After downgrading to the 2.2.0 this issue went away (with up to date versions of other software on the laptop). Command line arguments are the same.

**Comment by akallabeth** [comment]
@roussak ok, please add some more information (just as requested by the issue template), I can´t help here if I don´t even know if you run a desktop or a rails session.
Additionally, mouse events are passed through by `xfreerdp` (this is something that has not changed in ~10 years) so there might be something else interfering here, but again, please add some information we can work with here.

**Comment by roussak** [comment]
Command line I used is the following:
`xfreerdp /bpp:8 +compression -glyph-cache -offscreen-cache /w:1920 /h:1080 /floatbar /monitors:1 +auto-reconnect /u:user /v:hostname`
OS I used on my laptop is ArchLinux, architecture is x86_64.
About mouse events, I didn't understand you.

**Comment by akallabeth** [comment]
@roussak ok, weird command line. better use `/network:auto /size:1920x1080 +auto:reconnect`

As for mouse events, the code for that has not changed in a very long time, are you sure it has to do with the client version (=freerdp)?

**Comment by roussak** [comment]
I just tried your variant. First, it has a typo: `+auto-reconnect` is right. Second, it is not better, because it is slower. And finally, I went to my variant via documentation and wiki (I don't remember, probably on the Arch site). And one more thing: I got tonns of the
`[WARN][com.freerdp.client.x11] - xf_lock_x11_:   [1] recursive lock from xf_sw_end_paint`
on the console. I am not sure, if this good or bad...

About mouse: I can't add something to the written. I thought, that the behaviour described was caused by something on the local or remote side, so I tried to do some "light" things alike reboot etc. But, today I worked a whole day remotely with old (2.2.0) version and I was enjoying on my work. )) Yesterday I worked with the new version -- that was inconvenient. All other conditions are the same between "today" and "yesterday".
